frontend/src/components/Collections/components/DeleteCollection/index.tsx

```diff
@@ -22,14 +22,13 @@ const DeleteCollection: FC<Props> = ({
   Button = RawButton,
   isRevision,
 }) => {
-  const [deleteMutation, { isLoading }] = useDeleteCollection(id);
+  const { mutateAsync: deleteMutation, isLoading } = useDeleteCollection(id);
   const router = useRouter();
 
   const handleDelete = async () => {
-    await deleteMutation(
-      { collectionID: id },
-      { onSuccess: () => router.push(ROUTES.MY_COLLECTIONS) }
```

Comment:
pretty crazy, it seems like `mutateAsync` is not invoking `onSuccess()` callback for whatever reason, even though it should support callbacks. So to get around the issue, I'm just awaiting `deleteMutation()` first, then push route!

Reply:
Could we add that as a comment?

Reply:
Oh yes great idea! Done!